### PR TITLE
changes to protobuf services for gRPC, few other fixes

### PR DESCRIPTION
**Pull request by tanmaykm**
- code generated for services needed a few changes to have gRPC work with it.
    - service name prefixed with package name
    - `call_method` made aware of ServiceDescriptor
- upgraded tests, some code clean up
- changes to make it work on Julia 0.6

### CODE REVIEW

File: test/services/testsvc.jl [hunk 1]
```diff
@@ -22,42 +23,42 @@ end
 close(channel::TestRpcChannel) = close(channel.sock)
 
 type SvcHeader
-    method::ASCIIString
+    method::String
```
Comment (tkelman) [inline test/services/testsvc.jl]: if you still support 0.4 you'll want this to be Compat.String or it'll throw a depwarn

Comment (tanmaykm) [inline test/services/testsvc.jl]: Yes. thanks.
Should be fixed in https://github.com/tanmaykm/ProtoBuf.jl/pull/85.
But maybe we can go ahead with the tagging?

Comment (tkelman) [inline test/services/testsvc.jl]: do you plan on re-tagging promptly?

Comment (tanmaykm) [inline test/services/testsvc.jl]: re-tagging now...

Comment (tanmaykm) [inline test/services/testsvc.jl]: done

**Pull request merged**